### PR TITLE
Team 2 - merge to team branch

### PR DESCRIPTION
**Pull request by The-AhmadAA**
Shortcut keys now update button visuals
ESC key now opens the pause menu (closing with ESC is still a bit buggy but doesn't break anthing)
Streamlined BuildInputComponent code
Added pause status in GameTime Service, along with getters and setters
Code smell clean up

### CODE REVIEW

**Review by shiv-0831 — APPROVED** [review]
looks good!

File: source/core/src/main/com/csse3200/game/components/maingame/MainGameDisplay.java [hunk 1]
```diff
@@ -234,21 +258,22 @@ public void clicked(InputEvent event, float x, float y) {
                         TowerType selected = ServiceLocator.getCurrencyService().getTower();
                         if (selected == towers.get(4)) {
                             ServiceLocator.getCurrencyService().setTowerType(null);
-                            tower5.setChecked(false);
+
+                            towerToggle(tower5);
```
Comment (AlasdairS4698737) [inline source/core/src/main/com/csse3200/game/components/maingame/MainGameDisplay.java:262]: Was this meant to be towerToggle(null)? It breaks the previous pattern I've seen

**Review by Mohamad11Dab — APPROVED** [review]

**Review by AlasdairS4698737 — COMMENTED** [review]
I think there might be a mistake on  line 262 in the MainGameDisplay.
I've seen a pattern of towerToggle(null), with towerToggle(tower#) on a later line.
But for tower number 5, there were two instances of towerToggle(tower5). I wanted to check if this is intentional, or a typo.

File: source/core/src/main/com/csse3200/game/components/maingame/MainGameDisplay.java (hunk 1 above)
Comment (The-AhmadAA) [reply source/core/src/main/com/csse3200/game/components/maingame/MainGameDisplay.java:262]: good spotting! This was indeed supposed to be towerToggle(null)

**Pull request merged**